### ISSUE TITLE
Web Service Sender only caches the main WSDL, and not any sub-WSDL imports

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3310]
When the Web Service Sender is creating a service, it always manually requests the WSDL and caches it to the local filesystem. Then at runtime JAX-WS will use the file URL instead of the HTTP/S URL. This works fine, except when the main WSDL also imports _other_ WSDLs. Only the first one is being requested, and any others are not being cached.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3409
Reporter: narupley
Created: 2014-08-15T17:49:27.000-0700

**Comment by rbeckman-nextgen** [comment]
Revision 7293: The Web Service Sender is now caching all WSDLs in the definition tree, not just the main WSDL. When the dispatch container is being created, the connector will recursively replace all import location attributes with file URLs, after manually requesting the dependency at that location.

Imported Comment. Original Details:
Author: narupley
Created: 2014-08-15T17:53:46.000-0700

**Comment by rbeckman-nextgen** [comment]
Even though it isn't part of this ticket there was some other changes to the behavior in 3.1.0 where now anytime the wsdl is pulled from http/s it will be cached. The behavior in 3.0.3 was that it would only cache the wsdl when the server required basic authentication. 

Confirmed that in 3.0.3 the wsdl was not being cached unless the server required basic auth.

Confirmed that in 3.1.0(rev 7284) the wsdl is cached but any dependencies in the wsdl were not being cached so they would still be fetched by JAX-WS. This could pose problems for example if the server required ssl client authentication then JAX-WS would use the default SSL Context and fail to fetch the files(Even though the SSL Manager was configured). Another issue is that if we cached the WSDL and the import tags had local schemaLocations then the cached wsdl would be invalid because it would have no way of going out and grabbing those dependencies.

Did some testing in 3.1.0(rev 7352) and it looks like the WSDL dependencies are not cached if they just use a relative path. The dispatch is successful without caching so I believe this is legal. Reopening this ticket and I will upload some files which show this. 


Imported Comment. Original Details:
Author: eduardoa
Created: 2014-09-10T11:45:18.000-0700
